ui/src/components/Publication/Creation/MainText.tsx

```diff
@@ -212,6 +231,22 @@ const MainText: React.FC = (): React.ReactElement | null => {
         updateReferences(references.filter((item) => item.id !== id));
     };
 
+    const editReference = async (reference: Interfaces.Reference) => {
+        setModalReferenceInfo(reference);
+        setEditReferenceModalVisible(true);
+    };
+
+    const saveEditReferenceChanges = (referenceID: string, referenceContent: string) => {
+        //Find index of specific object using findIndex method.
+        const objIndex = references.findIndex((reference) => reference.id === referenceID);
```

Comment:
Wouldn't be easier to just create a new array from the existing references and just replace the edited one? With map